### DISCUSSION TITLE
Feature Request: Attach Screenshots to Code Snippets

### DISCUSSION THREAD

**Discussion by TheNomadInOrbit** [pieces-app/support#129] [Ideas]
I'd like to suggest adding a feature to attach screenshots with code snippets.

As developers, we often deal with numerous similar snippets. Incorporating a visual element, such as screenshots, can help us spot the one we need faster because we process images quicker than text. This would make finding code easier, particularly for frameworks like Flutter and React.

Thank you for considering my request. I look forward to any improvements that make our development process smoother.

<img width="1920" alt="310480163-bc4c7428-d312-45c9-a33c-438c946975b2" src="https://github.com/pieces-app/support/assets/36538038/6879e075-862e-4f24-a205-2fe90db0bf7b">

**Comment by caleb-at-pieces** [comment]
Interesting idea! Can see how this would help a lot with front end developement.

**Comment by rosie-at-pieces** [comment]
I have heard this as well from users who are looking to add documentation like diagrams to their snippets

**Comment by randyorten97** [comment]
Any progress so far on this one?

**Reply by mark-at-pieces** [reply]
hey @randyorten97 no progress on this one just yet, we are currently dialed into a V2 of our Long term memory, + replaceing our auth service.

However the ability to associate 2 materials like a code snippet and an image is a requested feature we have head about alot so I will ensure that we add this to the road map for the new year!

**Reply by mark-at-pieces** [reply]
** that being said a temporary fix for now, all images have an attached code snippet to you click on the ico in the top right when you persist an image 

ie the button with the blue dot

<img width="1036" alt="Screenshot 2025-02-17 at 8 22 51 AM" src="https://github.com/user-attachments/assets/559c1a5c-b692-4469-be60-d57b9850f277" />
